### PR TITLE
Allow Algolia search for CDNSource

### PR DESCRIPTION
**Pull request by igor-makarov**
Complementary to https://github.com/CocoaPods/CocoaPods/pull/9046

### CODE REVIEW

**Comment by dnkoutso** [comment]
Really like how this change is shaping up!

**Comment by dnkoutso** [comment]
@igor-makarov is it ready for review?

**Comment by igor-makarov** [comment]
@dnkoutso Yeah! I think it's ready.

File: lib/cocoapods-core/cdn_source.rb [hunk 1]
```diff
@@ -219,14 +219,21 @@ def search(query)
...
     def search_by_name(query, full_text_search = false)
       if full_text_search
-        raise Informative, "Can't perform full text search, it will take forever"
+        require 'algoliasearch'
+        begin
+          algolia_result = algolia_search_index.search(query, :attributesToRetrieve => 'name')
+          names = algolia_result['hits'].map { |r| r['name'] }
+          names.map { |n| set(n) }.reject { |s| s.versions.compact.empty? }
+        rescue Algolia::AlgoliaError => e
+          raise Informative, "CDN: #{name} - Cannot perform full-text search because Algolia returned an error: #{e}"
```
Comment (dnkoutso) [inline lib/cocoapods-core/cdn_source.rb]: should we expose "Algolia returned an error"? Maybe "Server returned an error"?

**Comment by dnkoutso** [comment]
How concerned are we with Algolia uptime? It seems the indexer is not working now in cocoapods.org? Can we push for someone to fix it?

File: lib/cocoapods-core/cdn_source.rb (hunk 1 above)
Comment (igor-makarov) [reply lib/cocoapods-core/cdn_source.rb]: Let's say that once we have more than one implementation, this message will be easy to change 🙃

**Pull request merged**